RUFAS/routines/animal/ration/ration_driver.py

```diff
@@ -309,64 +326,580 @@ class RationReporter:
+        kg_fed : float
+            Kilograms of feed item in ration
+        feed_item_info : Dict
+            Dictionary of nutrient and energy information of feed item
+        ration_report : Dict
+            Dictionary of nutrient amount and concentrations
+        body_weight : float
+            Animal body weight in kg
+
+        Returns
+        -------
+        float
+            Phosphorus supply, kg TODO: check units
+
+        """
+        if feed_item_info["type"] == "Forage":
+            dP = 0.64
+        elif feed_item_info["type"] == "Conc":
+            dP = 0.70
+        elif feed_item_info["type"] == "Mineral":
+            dP = 0.80
+        else:
+            dP = 0.0
+        return dP * feed_item_info["phosphorus"] * 0.01 * kg_fed
+
+    @classmethod
+    def get_fat(cls, kg_fed, feed_item_info, ration_report, body_weight):
+        """
+        Returns fat supply of feed item
+
+        Parameters
+        ----------
+        kg_fed : float
+            Kilograms of feed item in ration
+        feed_item_info : Dict
+            Dictionary of nutrient and energy information of feed item
+        ration_report : Dict
+            Dictionary of nutrient amount and concentrations
+        body_weight : float
+            Animal body weight in kg
+
+        Returns
+        -------
+        float
+            Fat supply of feed item, g TODO: check units
+
+        """
+        fat_item = feed_item_info["EE"] * kg_fed
+        return fat_item
+
+    @classmethod
+    def get_fat_percentage(cls, kg_fed, feed_item_info, ration_report, body_weight):
+        """
+        Returns fat content percentage of feed item
+
+        Parameters
+        ----------
+        kg_fed : float
+            Kilograms of feed item in ration
+        feed_item_info : Dict
+            Dictionary of nutrient and energy information of feed item
+        ration_report : Dict
+            Dictionary of nutrient amount and concentrations
+        body_weight : float
+            Animal body weight in kg
+
+        Returns
+        -------
+        float
+            Dry matter fat percentage of feed item
+
+        """
+        fat_item_percentage = (feed_item_info["EE"] * kg_fed) / ration_report["nutrient_amount"]["dm"]
+        return fat_item_percentage
+
+    @classmethod
+    def get_forage_NDF(cls, kg_fed, feed_item_info, ration_report, body_weight):
+        """
+        Returns forage neutral detergent fiber content of feed item, kg
+
+        Parameters
+        ----------
+        kg_fed : float
+            Kilograms of feed item in ration
+        feed_item_info : Dict
+            Dictionary of nutrient and energy information of feed item
+        ration_report : Dict
+            Dictionary of nutrient amount and concentrations
+        body_weight : float
+            Animal body weight in kg
+
+        Returns
+        -------
+        float
+            Forage neutral detergent fiber content of feed item, TODO: check units
+
```

Comment:
```suggestion
        Calculates the forage neutral detergent fiber content of feed item.

        Parameters
        ----------
        kg_fed : float
            Kilograms of feed item in ration.
        feed_item_info : Dict
            Dictionary of nutrient and energy information of feed item.
        ration_report : Dict
            Dictionary of nutrient amount and concentrations.
        body_weight : float
            Animal body weight in kg.

        Returns
        -------
        float
            Forage neutral detergent fiber content of feed item, TODO: check units.

```
I think it is better to specify what the method does in the top level docstring, then be more specific about what is returned in the "Returns" section. Otherwise the top level docstring is super redundant.

Also same comment as above for the TODO here.